src/CodeGen_C.cpp

```diff
@@ -1487,7 +1487,7 @@ class ExternCallPrototypes : public IRGraphVisitor {
     }
 
     bool has_c_declarations() const {
-        return !c_externs.empty();
+        return !c_externs.empty() || !destructors.empty();
```

Comment:
This looks like a general fix outside of what this PR is doing. If so, worth a separate PR?

Reply:
Well, maybe -- it's a drive-by that was needed only because the changes to the `nested_externs` test caused a call to `halide_device_free_as_destructor` to be injected when building under `host-cuda`, but it failed to compile because the call in question had no prototype. I can try to make minor changes to pull this out separately.